types/glob/index.d.ts

```diff
@@ -33,9 +33,10 @@ declare namespace G {
         stat?: boolean;
         silent?: boolean;
         strict?: boolean;
-        cache?: { [path: string]: any /* boolean | string | string[] */ };
-        statCache?: { [path: string]: fs.Stats };
-        symlinks?: any;
+        cache?: { [path: string]: boolean | 'DIR' | 'FILE' | string[] };
+        statCache?: { [path: string]: false | { isDirectory(): boolean} | undefined };
```

Comment:
for non-existent files `false` is stored in the cache, for any other error `undefined` is stored

https://github.com/isaacs/node-glob/blob/8882c8fccabbe459465e73cc2581e121a5fdd25b/sync.js#L443-L466